### DISCUSSION TITLE
DIA-NN 1.9.1 docker container

### DISCUSSION THREAD

**Discussion by MichelMoser** [vdemichev/DiaNN#1181] [General]
For those interested in running DIA-NN 1.9.1 on a UNIX system: 

A docker container has been created and is available at : 
```
docker pull michelmoser/diann-1.9.1
```

Usage: 
```
docker run -v ${PWD}:/data/ diann-1.9.1:latest diann-linux
```

**Comment by aliibarry** [comment]
This is great, thanks! Test run works with singularity

**Reply by filonovd** [reply]
Hi @aliibarry 
 Was there anything specific you had to do to be able to run it with singularity? I can run it in docker, but in apptainer it gives me an error :(

**Reply by aliibarry** [reply]
I've been running it using `singularity exec --env LANG=C.UTF-8,LC_ALL=C.UTF-8 -B ${PWD}:/data /xxx/bin/diann-1.9.1.sif diann-linux --gen-spec-lib --fasta-search --predictor`

**Reply by filonovd** [reply]
Oh, wow. Adding '--env LC_ALL=C.UTF-8' flag solved the issue. Thanks a lot!

**Comment by rolivella** [comment]
FYI: https://github.com/vdemichev/DiaNN/issues/1202#issuecomment-2417108281

**Reply by filonovd** [reply]
Thanks. For some reason it worked just fine building my own docker image, but wasn't working with the one from the repo. 
Added ENTRYPOINT to the Dockerfile for the ease of use as well - 
```
ENTRYPOINT ["/diann-1.9.1/diann-linux"]
```
